tools/cli/helpers/checkEnvironment.js

```diff
@@ -52,3 +55,49 @@ export async function compareComposerVersion() {
 		process.exit( 1 );
 	}
 }
+
+/**
+ * Check for whether the CLI is being run from within a different monorepo checkout.
+ *
+ * If so, this will shell out to the correct CLI, then return.
+ */
+export async function checkCliLocation() {
+	// Did our caller already do this? Don't check again.
+	if ( process.env.JETPACK_CLI_DID_REEXEC ) {
+		return;
+	}
+
+	// Use `path.dirname()` to ensure same trailing-slash behavior as is used below.
+	const thisRoot = path.dirname( fileURLToPath( new URL( '../../', import.meta.url ) ) );
+
+	for (
+		let olddir = null, dir = process.cwd();
+		dir !== olddir;
+		olddir = dir, dir = path.dirname( dir )
+	) {
+		const exe = path.join( dir, 'tools/cli/bin/jetpack.js' );
+		if ( ! fs.existsSync( exe ) ) {
+			continue;
+		}
+
+		if ( dir === thisRoot ) {
+			return;
+		}
+
+		console.log(
+			chalk.yellow(
+				`Jetpack CLI was linked to ${ thisRoot }, but a Jetpack Monorepo checkout was found at ${ dir }. Executing the CLI from there instead.`
```

Comment:
A really minor suggestion, but the wording somehow threw me and I thought the CLI was being run from the first mentioned path. I've re-read it a few times and it does make sense, but I thought a super clear wording change might avoid any initial confusion like I had. What do you think?

`Jetpack CLI was linked to ${ thisRoot }, but a Jetpack Monorepo checkout was found at ${ dir }. Executing the CLI from ${ dir }.`

Reply:
Done in 3c10322